### PR TITLE
[#20][UI] As a user, I can see confirmation modal before leaving the survey

### LINKED ISSUE #20: [UI] As a user, I can see confirmation modal before leaving the survey
## Why
The application supports various types of survey items. After the user fills in all required survey items, the survey can be submitted and mark the survey as completed.

## Acceptance Criteria
- Display confirmation modal when user clicks on exit button.
image

- Redirect the user to the survey dashboard if the user confirms the modal.

### PR DESCRIPTION
**Pull request by hanam1ni**
- Closes #20 

## What happened 👀

- Add a close button to the question page.
- Show confirmation modal when clicking on a close button.
 
## Proof Of Work 📹

https://user-images.githubusercontent.com/14077479/179799497-b82c3848-3274-40ed-86c9-a32bcaab5083.mov

### CODE REVIEW

File: src/components/Button/Button.module.css [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
 }
 
 .primary {
   @apply bg-white text-black hover:bg-indigo-800 hover:text-white 
     focus:bg-indigo-900 focus:text-white;
 }
+
+.secondary {
+  @apply bg-gray-700 text-white hover:bg-indigo-800 hover:text-white 
+  focus:bg-indigo-900 focus:text-white;
```
Comment (malparty) [inline src/components/Button/Button.module.css:13]:
```suggestion
  @apply bg-gray-700 text-white hover:bg-indigo-800 hover:text-white 
    focus:bg-indigo-900 focus:text-white;
```
Should the indentation be higher on that second line? 🤔

File: src/components/Modal/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      const { getByRole, getByTestId } = render(
+        <Modal show={false}>
+          <Greeting />
+        </Modal>
+      );
+
+      expect(getByTestId('modal-overlay')).toHaveClass('opacity-0 -z-50');
+      expect(getByRole('dialog')).toHaveClass('opacity-0 -z-50');
+    });
+  });
```
Comment (malparty) [inline src/components/Modal/index.test.tsx:34]: Should we add 2 tests:
- When `show: true`, it displays the provided content
- When `show: false`, it hides the provided content (= the text inside `<Greeting />`)

File: src/pages/surveys/[id]/questions.module.css [hunk 3]
```diff
@@ -3,6 +3,7 @@
   cursor-pointer hover:bg-gray-400 transition-all duration-200;
 }
 
-.nextQuestionLink:disabled {
-  @apply opacity-50 cursor-not-allowed hover:bg-white;
+.leaveSurveyButton {
+  @apply w-7 h-7 absolute top-8 right-8 z-10 flex justify-center items-center bg-gray-400 rounded-full
+  cursor-pointer hover:bg-gray-600 transition-all duration-200;
```
Comment (malparty) [inline src/pages/surveys/[id]/questions.module.css:8]:
```suggestion
    cursor-pointer hover:bg-gray-600 transition-all duration-200;
```
Same indentation missing here?

**Review by malparty — CHANGES_REQUESTED** [review]
A few small suggestions :)

File: src/components/Button/Button.module.css (hunk 1 above)
Comment (hanam1ni) [reply src/components/Button/Button.module.css]: Updated in [e38244a](https://github.com/hanam1ni/react-surveyor/pull/93/commits/e38244a4df736e21c5156fc7a53850187611f59d)

File: src/pages/surveys/[id]/questions.module.css (hunk 3 above)
Comment (hanam1ni) [reply src/pages/surveys/[id]/questions.module.css]: Updated in [e38244a](https://github.com/hanam1ni/react-surveyor/pull/93/commits/e38244a4df736e21c5156fc7a53850187611f59d)

File: src/components/Modal/index.test.tsx (hunk 2 above)
Comment (malparty) [reply src/components/Modal/index.test.tsx:34]: I'm not sure, but from my understanding, the tests would still PASS, even if I remove the `{children}` part of the Modal component. So an empty modal bug would pass through the tests, no? 🤔

Comment (hanam1ni) [reply src/components/Modal/index.test.tsx:36]: @malparty I updated in this commit [c5335a7](https://github.com/hanam1ni/react-surveyor/pull/93/commits/c5335a71735fdc379475ef6084f8fa5ff4f490cb). Now the modal test will check if children are rendered by text searching and visibility of the modal by classes checking

**Review by malparty — APPROVED** [review]

**Review by hoangmirs — APPROVED** [review]

**Pull request merged**